### PR TITLE
Full-text search within an index example (#220)

### LINKED ISSUE #220: Hacktoberfest Examples: Full-Text Search within an Index
It would be pretty cool to see some examples of how to use Full-Text search within Redis OM.NET indexes, these mostly need to be decorated with `Text` attributes, and then more or less queried as normal. For more information on how to do full-text search in RediSearch (and probably some inspiration for this topic) check out the RediSearch docs, probably the topic on [FT.SEARCH](https://redis.io/commands/ft.search/) will be most instructive.

See #217 before getting started.

If you’d like to work on this please mention me [@slorello89](https://github.com/slorello89) or [@SuzeShardlow](https://github.com/SuzeShardlow) in a comment and we’ll assign this issue to you.
This is part of our participation in Hacktoberfest 2022. Learn how to complete this issue here: https://redis.io/community/hacktoberfest/
Need help or want to talk to us? Join us on Discord where we have a dedicated Hacktoberfest [channel](https://discord.com/channels/697882427875393627/895802042797395989)

### PR DESCRIPTION
**Pull request by Jeevananthan-23**
* Adding Full-text search within an index examples
*  Closes #220.
*  My Discard handles BadeyeJeeva#5477. Thanks!

### CODE REVIEW

**Comment by rtynski** [comment]
My suggestions:
 - Full-Text_Search_within_an_Index to PascalCase (FullTextSearchWithinAnIndex)
 - RedisHelper to Helpers (as in Redis.OM.BasicMatchingQueries)
 - .sln file to delete (unspecifed @slorello89)

File: examples/Redis.OM.Full-Text_Search_within_an_Index/Program.cs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+            // Create index
+            connection.CreateIndex(typeof(Movie));
+            connection.CreateIndex(typeof(Awards));
+
+            var redisHelper = new RedisHelper(provider);
+            redisHelper.InitializeCustomers();
+
+            //BasicFirstOrDefaultQuery
+            // get first or default null value from movies redis limit set to 0 to 1
+            var movieWithFirstorDefault = _movieCollection.FirstOrDefault<Movie>(x => x.ImdbRating >5.5);
```
Comment (rtynski) [inline examples/Redis.OM.Full-Text_Search_within_an_Index/Program.cs:29]: camelCase: movieWithFirstOrDefault

File: examples/Redis.OM.Full-Text_Search_within_an_Index/Program.cs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            Console.WriteLine($"Title of the movie:{movieWithFirstorDefault.Title} and ImdbRating of the movie is {movieWithFirstorDefault.ImdbRating}");
+
+            //BasicQueryWithExactNumericMatch
+            //Get the list of movies which year's is eq to 2003
+            var getListofMovie = _movieCollection.Where(x => x.Year == 2003).ToList();
```
Comment (rtynski) [inline examples/Redis.OM.Full-Text_Search_within_an_Index/Program.cs:34]: camelCase: getListOfMovie

File: examples/Redis.OM.Full-Text_Search_within_an_Index/Program.cs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            ShowMovies("Year is 2003", getListofMovie);
+
+            //QueryWithNegation
+            var negationMovieList = _movieCollection.Where( x => x.Runtime >= 178 && x.ImdbRating!=8.0).ToList();
+            ShowMovies("Negation in MovieList", negationMovieList);
+
+            //FirstOrDefaultWithMixedLocals
+            var rating = 8.8;
+            var searchmixedLocals = new Movie();
```
Comment (rtynski) [inline examples/Redis.OM.Full-Text_Search_within_an_Index/Program.cs:43]: camelCase: searchMixedLocals

File: examples/Redis.OM.Full-Text_Search_within_an_Index/Program.cs [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+            //search with contains
+            var oscarsWinningMovies = _movieCollection.Where(x => x.Awards.Text.Contains("Oscars")).ToList();
+            ShowMovies("won Oscars", oscarsWinningMovies);
+
+            //ListContains 
+            var searchForListOfCastInMovie = _movieCollection.Where(x => x.Cast.Contains("Sean Astin")).ToList();
+            ShowMovies("Cast By Sean Astin", searchForListOfCastInMovie);
+
+            //Sort the search result ascending use OrdreByDecending for DESC
```
Comment (rtynski) [inline examples/Redis.OM.Full-Text_Search_within_an_Index/Program.cs:75]: typo: OrderByDescending

File: examples/Redis.OM.Full-Text_Search_within_an_Index/Models/Movie.cs [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        public List<Genres>  Genres { get; set; }
+        [Indexed(CascadeDepth = 2)]
+        public Awards Awards { get; set; }
+        [Indexed(Sortable =true)]
+        public double? ImdbRating { get; set; }
+        [Indexed]
+        public Country Country { get; set; }
+
+    }
+    public enum Country
```
Comment (rtynski) [inline examples/Redis.OM.Full-Text_Search_within_an_Index/Models/Movie.cs:37]: Separate file

**Comment by Jeevananthan-23** [comment]
Waiting for @slorello89 feedback so, I can make changes. Thanks!

File: examples/Redis.OM.Full-Text_Search_within_an_Index/Models/Awards.cs [hunk 6]
```diff
@@ -0,0 +1,17 @@
+﻿
+
+using Redis.OM.Modeling;
+
+namespace Redis.OM.Full_Text_Search_within_an_Index.Models
+{
+    [Document(IndexName = "awards-idx", StorageType = StorageType.Json)]
+    public class Awards
+    {
+        [Indexed(Sortable =true)]
```
Comment (slorello89) [inline examples/Redis.OM.Full-Text_Search_within_an_Index/Models/Awards.cs:10]:
```suggestion
        [Indexed(Sortable = true)]
```

File: examples/Redis.OM.Full-Text_Search_within_an_Index/Models/Awards.cs [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+        public int Wins { get; set; }
+        [Indexed(Sortable =true)]
```
Comment (slorello89) [inline examples/Redis.OM.Full-Text_Search_within_an_Index/Models/Awards.cs:12]:
```suggestion
        [Indexed(Sortable = true)]
```

File: examples/Redis.OM.Full-Text_Search_within_an_Index/Program.cs [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+
+            //BasicFirstOrDefaultQuery
+            // get first or default null value from movies redis limit set to 0 to 1
+            var movieWithFirstorDefault = _movieCollection.FirstOrDefault<Movie>(x => x.ImdbRating >5.5);
+            Console.WriteLine($"Title of the movie:{movieWithFirstorDefault.Title} and ImdbRating of the movie is {movieWithFirstorDefault.ImdbRating}");
+
+            //BasicQueryWithExactNumericMatch
+            //Get the list of movies which year's is eq to 2003
+            var getListofMovie = _movieCollection.Where(x => x.Year == 2003).ToList();
+            ShowMovies("Year is 2003", getListofMovie);
```
Comment (slorello89) [inline examples/Redis.OM.Full-Text_Search_within_an_Index/Program.cs:35]:
```suggestion
            ShowMovies("Year is 2003", getListOfMovies);
```

**Review by slorello89 — APPROVED** [review]
looking pretty good @Jeevananthan-23 - couple minor items and then I think it's good to go.

**Review by slorello89 — CHANGES_REQUESTED** [review]
accidentally hit approve :)

**Review by Jeevananthan-23 — COMMENTED** [review]
Hi @slorello89, changes made please look into it.

**Comment by slorello89** [comment]
Hi @Jeevananthan-23 - I noticed some quirks in the namespaces (and the file/folder names) so I made some quick edits to those - should be good to go after this :)

**Review by slorello89 — APPROVED** [review]
LGTM 👍 thanks for the contribution! happy 🎃 Hacktoberfest 🎃

**Pull request merged**